src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/KeyboardInt16Handler.cs

```diff
@@ -62,8 +59,89 @@ private void Unsupported(int operation) {
     /// <inheritdoc/>
     public override byte VectorNumber => 0x16;
 
+    /// <summary>
+    ///     Emits a minimal INT 16h handler stub into guest RAM that handles AH=00h (GetKeystroke) in-place
+    ///     and dispatches all other AH values back to the C# handler via a callback.
+    /// </summary>
+    /// <remarks><![CDATA[
+    ///     Assembled control flow (labels for readability):
+    ///     L_HANDLER:
+    ///     cmp ah, 00h
+    ///     jz  L_AH00
+    /// 
+    ///     L_DEFAULT:
+    ///     int 0A4h   ; DefaultDispatch -> calls C# Run()
+    ///     iret
+    /// 
+    ///     L_AH00:
+    ///     int 0A0h   ; CallbackHasKey -> sets ZF=0 if a key is present
+    ///     jnz have_key
```

Comment:
jnz have_key => so it will jump if ZF = 1 which means if "_biosKeyboardBuffer.IsEmpty"

Is the label right? Is it me interpreting that wrongly?